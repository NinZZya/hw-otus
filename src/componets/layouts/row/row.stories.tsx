import React from "react";
/** @jsx jsx */
import { jsx } from "@emotion/core";
// also exported from "@storybook/react" if you can deal with breaking changes in 6.1
import { Meta } from "@storybook/react/types-6-0";

import { Row } from "./row";

export default {
  title: "Example/layouts/Row",
  component: Row,
} as Meta;

const style = { padding: "1rem", border: "1px dotted #000" };

export const RowWithElements: React.FC = () => (
  <Row style={style}>
    <p css={style}>Item 1</p>
    <p css={style}>Item 2</p>
  </Row>
);
