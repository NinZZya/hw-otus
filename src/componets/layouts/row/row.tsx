import React, { Children, FC } from "react";
/** @jsx jsx */
import { jsx, css } from "@emotion/react";
import { SerializedStyles } from "@emotion/utils/types";

interface RowProp {
  [key: string]: string | undefined | Element[] | React.ReactNode[];
  justifyContent?: string;
  alignItems?: string;
  margin?: string;
  padding?: string;
}

const DefaultStyle = {
  JUSTIFY_CONTENT: "space-around",
  ALIGIN_ITEMS: "center",
  MARGIN: "0",
  PADDING: "1rem",
};

const getRowStyle = (style: RowProp): SerializedStyles => css`
  display: flex;
  flex-direction: column;
  justify-content: ${style.justifyContent
    ? style.justifyContent
    : DefaultStyle.JUSTIFY_CONTENT};
  align-items: ${style.alignItems
    ? style.alignItems
    : DefaultStyle.ALIGIN_ITEMS};
  width: 100%;
  height: 100%;
  margin: ${style.margin ? style.margin : DefaultStyle.MARGIN};
  padding: ${style.padding ? style.padding : DefaultStyle.PADDING};
  box-sizing: border-box;
`;

export const Row: FC<RowProp> = (props) => {
  return (
    <div css={getRowStyle(props)}>
      {Children.map(props.children, (child) => child)}
    </div>
  );
};
