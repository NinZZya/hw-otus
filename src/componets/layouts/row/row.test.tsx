import React from "react";
import renderer from "react-test-renderer";
import { Row } from "./row";

describe(`Row render coorectly`, () => {
  it(`Row without props render coorectly`, () => {
    const tree = renderer.create(<Row />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Row with components render coorectly`, () => {
    const tree = renderer
      .create(
        <Row>
          <div>div</div>
          <div>div</div>
          <div>div</div>
        </Row>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
