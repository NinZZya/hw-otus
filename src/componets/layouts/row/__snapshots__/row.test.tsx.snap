// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row render coorectly Row with components render coorectly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "1ppfip7",
      "next": undefined,
      "styles": "
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 1rem;
  box-sizing: border-box;
",
      "toString": [Function],
    }
  }
>
  <div>
    div
  </div>
  <div>
    div
  </div>
  <div>
    div
  </div>
</div>
`;

exports[`Row render coorectly Row without props render coorectly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "1ppfip7",
      "next": undefined,
      "styles": "
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 1rem;
  box-sizing: border-box;
",
      "toString": [Function],
    }
  }
/>
`;
